packages/core/src/sync-store/migrations.ts

```diff
@@ -832,6 +832,157 @@ const migrations: Record<string, Migration> = {
         .execute();
     },
   },
+  "2024_09_09_0_adjacent_interval": {
+    async up(db: Kysely<any>) {
+      await db.schema
+        .createTable("interval")
+        .addColumn("id", "serial", (col) => col.notNull().primaryKey()) // Auto-increment
+        .addColumn("fragment_id", "text", (col) => col.notNull())
+        .addColumn("chain_id", "integer", (col) => col.notNull())
+        .addColumn("start_block", "numeric(78, 0)", (col) => col.notNull())
+        .addColumn("end_block", "numeric(78, 0)", (col) => col.notNull())
+        .execute();
+
+      await db
+        .insertInto("interval")
+        .columns(["fragment_id", "chain_id", "start_block", "end_block"])
+        .expression(
+          db
+            .selectFrom("logFilters as lf")
+            .innerJoin("logFilterIntervals as lfi", "lf.id", "lfi.logFilterId")
+            .select([
+              "lf.id as fragment_id",
+              "lf.chainId as chain_id",
+              "lfi.startBlock as start_block",
+              "lfi.endBlock as end_block",
+            ]),
+        )
+        .execute();
+
+      await db.schema.dropTable("logFilters").ifExists().cascade().execute();
+      await db.schema
+        .dropTable("logFilterIntervals")
+        .ifExists()
+        .cascade()
+        .execute();
+
+      await db
+        .insertInto("interval")
+        .columns(["fragment_id", "chain_id", "start_block", "end_block"])
+        .expression(
+          db
+            .selectFrom("factoryLogFilters as flf")
+            .innerJoin(
+              "factoryLogFilterIntervals as flfi",
+              "flf.id",
+              "flfi.factoryId",
+            )
+            .select([
+              "flf.id as fragment_id",
+              "flf.chainId as chain_id",
+              "flfi.startBlock as start_block",
+              "flfi.endBlock as end_block",
+            ]),
+        )
+        .execute();
+
+      await db.schema
+        .dropTable("factoryLogFilters")
+        .ifExists()
+        .cascade()
+        .execute();
+      await db.schema
+        .dropTable("factoryLogFilterIntervals")
+        .ifExists()
+        .cascade()
+        .execute();
+
+      await db
+        .insertInto("interval")
+        .columns(["fragment_id", "chain_id", "start_block", "end_block"])
+        .expression(
+          db
+            .selectFrom("traceFilters as tf")
+            .innerJoin(
+              "traceFilterIntervals as tfi",
+              "tf.id",
+              "tfi.traceFilterId",
+            )
+            .select([
+              "tf.id as fragment_id",
```

Comment:
While we're doing a migration, I think it would be worthwhile to document each of these fragment ID "schemas", and perhaps think ahead towards allowing factory configs to be included in more than just `log.address` and `trace.toAddress`, as specified [here](https://github.com/ponder-sh/ponder/issues/1198). Maybe they are already future-proof, not sure.

These sync store database migrations are "expensive" (currently a one-way door for a given database) so we should aim to minimize the # required.